simulation/test_swarm.cpp

```diff
@@ -537,6 +538,56 @@ TORRENT_TEST(torrent_completed_alert)
 	TEST_EQUAL(num_file_completed, 1);
 }
 
+TORRENT_TEST(block_uploaded_alert)
+{
+	// blocks[piece count][number of blocks per piece] (each block's element will
+	// be set to true when a block_uploaded_alert alert is received for that block)
+	std::vector<std::vector<bool>> blocks;
+
+	setup_swarm(2, swarm_test::upload
+		// add session
+		, [](lt::settings_pack& pack)
+		{
+			pack.set_int(lt::settings_pack::alert_mask,
+				alert::progress_notification | alert::status_notification);
+		}
+		// add torrent
+		, [](lt::add_torrent_params&) {}
+		// on alert
+		, [&](lt::alert const* a, lt::session&) {
+			if (auto at = lt::alert_cast<lt::add_torrent_alert>(a))
+			{
+				// init blocks vector, MUST happen before any block_uploaded_alert alerts
+				std::shared_ptr<lt::torrent> torrent = at->handle.native_handle();
+				std::shared_ptr<const torrent_info> torrent_info = torrent->get_torrent_copy();
+				int blocks_per_piece = torrent_info->piece_length() / torrent->block_size();
+				for (int i = 0; i < torrent_info->num_pieces(); ++i)
```

Comment:
I believe you could simplify this loop to:
``blocks.resize(torrent_info->num_pieces(), std::vector<bool>(blocks_per_piece, false));``